WikiWatershed/model-my-watershed
# Add Tests for New Geoprocessing API Endpoints

## Overview

Adds test for the new Geoprocessing API endpoints.

Most of these tests run the unsuccessful cases, because success runs require a large amount of mocking, from Celery to Geoprocessing Service to Data Tables that are not available in the test database.

There is also a test for applying modifications to existing GMS payloads. This is also not exhaustive, but demonstrative, and covers basic use.

Also improves the performance of some tests by switching from `LiveServerTestCase` to `TestCase`.

Connects #3511 

## Testing Instructions

* Check out this PR
* Run `./scripts/test.sh`
  - [x] Ensure it finishes successfully
